### ISSUE TITLE
[BUG] Running U3 Gates on IonQ qBraid Provider

### ISSUE THREAD

**Issue by rryoung98** [qBraid/qBraid#840]
### Environment - Output of qbraid.about()

```
The IonQ Provider JSON/ qBraid SDK needs to transpire the U3 gate to successfully run many of the Classiq engine QASM files
```

### Context - Running in qBraid Lab?

Yes

### What happened?

When using the IonQ Provider submitting the QASM file with U3 gates does not run. (I sent a test file to you @ryanhill1 )

### Suggestions (Optional)

Converting to qiskit first then to dumping it to qasm3 to pass through a synthesis function

**Comment by ryanhill1** [comment]
I didn't get the QASM file, where'd you send it to? But problem actually sounds like gate synthesis not decomposition. You can't "decompose" any further than U3, that's the lowest you can go. But basically you would have to synthesize these U3 gates to create other gates that are in the IonQ standard gate set. Or, if the input program doesn't originally contain U3 gates, you would need to do some sort of custom qasm conversion that does not include U3 gates. Tough part is, many of the qiskit gates by default use U3 in the QASM representations.

**Comment by rryoung98** [comment]
ah yes you will receive the file in the morning tomorrow 😅 , timed message.

**Comment by ryanhill1** [comment]
Yeah converting this .qasm file to the IonQ gate set isn't something the qBraid-SDK can currently do. Your best bet it to use either:

`qiskit.compiler.transpile`: https://quantumcomputing.stackexchange.com/a/17861/13991
`pytket.passes.RebaseCustom`: https://quantumcomputing.stackexchange.com/a/31505/13991
`cirq.optimize_for_target_gateset`: https://quantumcomputing.stackexchange.com/a/24478/13991

or installing `qiskit-ionq` and using `qiskit.transpile` with the IonQ backend: https://docs.ionq.com/sdks/qiskit/native-gates-qiskit#transpiling-a-circuit-to-native-gates

or something similar for `cirq-ionq` or `pytket-ionq`
